ethcore/light/src/client/mod.rs

```diff
@@ -109,12 +107,12 @@ impl Client {
 
 	/// Fetch a vector of all pending transactions.
 	pub fn ready_transactions(&self) -> Vec<PendingTransaction> {
-		let best_num = self.chain.best_block().number;
+		let best = self.chain.best_header();
 		self.tx_pool.lock()
 			.values()
 			.filter(|t| match t.condition {
-				Some(TransactionCondition::Number(ref x)) => x <= &best_num,
-				Some(TransactionCondition::Timestamp(ref x)) => *x <= time::get_time().sec as u64,
+				Some(TransactionCondition::Number(x)) => x <= best.number(),
+				Some(TransactionCondition::Timestamp(x)) => x <= best.timestamp(),
```

Comment:
Pending transactions might be first propagated a little bit too late (we need to first import the block with given timestamp and only after that the propagation starts).
It's probably OK, but we need to make sure to indicate this in docs/ui so that people understand the consequences.